spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,227 @@
+require_relative "spec_helper"
+
+describe "ReservationManager class" do
+  describe "#initialize" do
+    let (:manager) { Hotel::ReservationManager.new }
+
+    it "must be an instance of ReservationManager" do
+      expect(manager).must_be_kind_of Hotel::ReservationManager
+    end
+
+    it "must create an array of all rooms in the hotel" do
+      expect(manager.rooms).must_be_kind_of Array
+      (0..19).each { |num| expect(manager.rooms[num]).must_be_kind_of Hotel::Room }
+      assert_nil(manager.rooms[20], msg = nil)
+    end
+
+    it "creates an empty array for reservations" do
+      expect(manager.reservations).must_be_kind_of Array
+    end
+  end
+
+  describe "#request_reservation" do
+    let (:manager) { Hotel::ReservationManager.new }
+    let (:new_reservation) { manager.request_reservation("Feb12", "Feb15", booking_name: "Kim") }
+
+    it "adds a reservation to the reservations array" do
+      before_res = manager.reservations.length
+      new_reservation
+      expect(manager.reservations.length).must_equal before_res + 1
+    end
+
+    it "accurately loads a reservation into the reservations array" do
+      new_reservation
+      first_reservation = manager.reservations.first
+      expect(first_reservation.check_in).must_equal Date.parse("Feb12")
+      expect(first_reservation.check_out).must_equal Date.parse("Feb15")
+      expect(first_reservation.all_dates).must_be_kind_of Array
+      expect(first_reservation.all_dates.length).must_equal 3
+      expect(first_reservation.all_dates[0]).must_equal Date.parse("Feb12")
+      expect(first_reservation.room.number).must_be :<=, 20
+      expect(first_reservation.room.number).must_be :>, 0
+      assert_nil(first_reservation.block_name, msg = nil)
+    end
+
+    it "adds the reservation to the room" do
```

Comment:
You should also test it with invalid dates and that you can reserve room on the same day the previous reservation ends.